odoo/addons/base/models/ir_mail_server.py

```diff
@@ -382,6 +385,32 @@ def build_email(self, email_from, email_to, subject, body, email_cc=None, email_
                 msg.attach(part)
         return msg
 
+    def _set_email_from(self, msg, headers, email_from):
```

Comment:
I don't like methods that update an existing object as a side-effect. I would prefer to have a ``_get_email_from`` that returns the email_from and caller updates msg. That way this method only computes stuff, and caller makes whatever he wants with it. For ``Return-Path`` it seems caller may update it only if computed_email_from is different from email_from, isn't it ?